File: Medium/Java/62. Unique Paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[] row = new int[n];
        for(int i = 0; i < row.length; i++){
            row[i] = 1;
        }
        
        for(int i = 1; i < m; i++){
            int[] currRow = new int[n];
            currRow[0] = 1;
            for(int j = 1; j < n; j++){
                currRow[j] = 1;
                currRow[j] = currRow[j - 1] + row[j];
            }
            row = currRow;
        }

        return row[n - 1];
    }
}

File: Medium/Python/62. Unique Paths.py
class Solution(object):
    def uniquePaths(self, m, n):
        row = [1] * n

        for i in range(1, m):
            currRow = [1] * n
            for j in range(1, n):
                currRow[j] = currRow[j - 1] + row[j]
            row = currRow
        
        return row[-1]
